### PR TITLE
Pass the host route IP family to the guest 

### PR DESCRIPTION
**Pull request by sameo**
So that we can properly select the same IP family when setting and updating routes in the guest.

Fixes: #2768

### CODE REVIEW

**Comment by fidencio** [comment]
Although this sounds intrusive for a backport, I think it's needed & worth it.  Could someone from @kata-containers/redhat review this one?

**Comment by sameo** [comment]
> Although this sounds intrusive for a backport, I think it's needed & worth it. Could someone from @kata-containers/redhat review this one?

Right, I will backport it once it's approved and CI green.

File: src/agent/src/netlink.rs [hunk 1]
```diff
@@ -312,7 +312,11 @@ impl Handle {
 
         for route in list {
             let link = self.find_link(LinkFilter::Name(&route.device)).await?;
-            let is_v6 = is_ipv6(route.get_gateway()) || is_ipv6(route.get_dest());
+            let is_v6 = if route.get_gateway().is_empty() && route.get_dest().is_empty() {
+                route.get_family() == IPFamily::v6
```
Comment (egernst) [inline src/agent/src/netlink.rs:316]: Is there any reason we wouldn't want to use get_family directly, regardless if the gateway and destination are not provided?  Are we populating the family as part of the gRPC regardless?

Comment (sameo) [reply src/agent/src/netlink.rs:316]: This is for backward compatibility purposes, unless we're ok with saying the new agent will only work with the corresponding  runtime changes.

File: src/runtime/cmd/netmon/netmon.go [hunk 2]
```diff
@@ -170,7 +170,7 @@ func newNetmon(params netmonParams) (*netmon, error) {
 
 func (n *netmon) cleanup() {
 	os.RemoveAll(n.storagePath)
-	n.netHandler.Delete()
```
Comment (c3d) [inline src/runtime/cmd/netmon/netmon.go:173]: Is this a change in the interface? If so, could you announce it in the commit message to explain the change?

**Review by c3d — APPROVED** [review]
Looks good to me.

Do you have a test case? (I did not see it)

File: src/runtime/cmd/netmon/netmon.go (hunk 2 above)
Comment (sameo) [reply src/runtime/cmd/netmon/netmon.go:173]: Good point. I added a note on the commit message now.

File: src/agent/src/netlink.rs (hunk 1 above)
Comment (sameo) [reply src/agent/src/netlink.rs]: @egernst I changed my mind and went the cleaner way: We only rely on the Route family field now for switching between v4 and v6. Running this new agent means you'll have to run the fixed runtime as well, but I believe this is acceptable.

**Review by Jakob-Naucke — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @sameo.

lgtm. Since @egernst has commented, maybe we should hold off merging until he's acked this too.

File: src/runtime/go.sum [hunk 3]
```diff
@@ -532,8 +532,9 @@ github.com/urfave/cli v1.22.1/go.mod h1:Gos4lmkARVdJ6EkW0WaNv/tZAAMe9V7XWyB60NtX
 github.com/urfave/cli v1.22.2 h1:gsqYFH8bb9ekPA12kRo0hfjngWQjkJPlN9R0N78BoUo=
 github.com/urfave/cli v1.22.2/go.mod h1:Gos4lmkARVdJ6EkW0WaNv/tZAAMe9V7XWyB60NtXRu0=
 github.com/vishvananda/netlink v1.1.0/go.mod h1:cTgwzPIzzgDAYoQrMm0EdrjRUBkTqKYppBueQtXaqoE=
-github.com/vishvananda/netlink v1.1.1-0.20201029203352-d40f9887b852 h1:cPXZWzzG0NllBLdjWoD1nDfaqu98YMv+OneaKc8sPOA=
 github.com/vishvananda/netlink v1.1.1-0.20201029203352-d40f9887b852/go.mod h1:twkDnbuQxJYemMlGd4JFIcuhgX83tXhKS2B/PRMpOho=
```
Comment (gkurz) [inline src/runtime/go.sum:535]: Shouldn't the go.mod line be removed all the same ?

Comment (sameo) [reply src/runtime/go.sum:535]: FWIW, this was autogenerated for me by `go mod`. I'm not sure I want to manually modify that file...

Comment (gkurz) [reply src/runtime/go.sum:535]: Indeed ! And if you remove it, `go mod tidy` consistently puts it back. Thanks, that answers my question.

**Pull request merged**